scripts/amd/gemm/README.md

```diff
@@ -1,44 +1,164 @@
+- split-k algorithm, which is controled by `SPLIT_K`.
+- Bias along M dim, which is controled by `BIAS` and `bias_ptr`.
+- Masked load along K dim inside the loop, which is controled by `EVEN_K`.
+This means `BLOCK_SIZE_K` does not need to divide K dim.
 
-### Usage
-Go to the script dir
-```bash
-cd triton/scripts/amd/gemm/
+### Differences between the tutorial
+
+Unlike the [matmul tutorial](https://github.com/triton-lang/triton/blob/main/python/tutorials/03-matrix-multiplication.py) (referred as the tutorial), 
+the matmul kernel used in the tuning script (referred as the kerel) does not 
+guard load along M and N dim 
+([this](https://github.com/triton-lang/triton/blob/main/python/tutorials/03-matrix-multiplication.py#L282-L283) shows how this is done in the tutorial).
+When `BLOCK_SIZE_M` or `BLOCK_SIZE_N` does not divide M or N, the kernel will 
+load out-of-bound data.
+In most cases this is fine, since the kernel does masked store at the end.
+However, this may lead to GPU memory access fault in some cases, especially
+when the tensor is large. 
+We will fix this issue in the future.
+
+
+## Tuning script usage
+
+### Tuning mode
+
+The tuning script can take one or more gemm sizes and run tuning for them.
+The input gemm sizes are prepared in a yaml file. Here is an example yaml file:
+```yaml
+- {'M': 4864, 'N': 4096, 'K': 8256, 'rowMajorA': 'T', 'rowMajorB': 'N'}
+- {'M': 512, 'N': 512, 'K': 512, 'rowMajorA': 'T', 'rowMajorB': 'N'}
 ```
 
-1. Tune gemm sizes given in a yaml file and check correctness on the way
-```bash
-python tune_gemm.py --gemm_size_file input_gemm_sizes.yaml --compare
+The tuning script works as follows
+```python
+./tune_gemm --gemm_size_file input.yaml [options]
+```
+The following `options` are supported in the tuning mode
+
+- Input data types:
+  - `-dtype_a dtype`, `-dtype_b dtype`, and `-dtype_c dtype`: input and output element type.
+  - Supported `dtype`: fp16 (default), bf16, fp8, bf8, int8, int32, fp32
+- Parallel compilation of kernels:
+  - `num_threads n` controls that n threads will
+  be used in the compilation stage. The default value is 32.
+  - `--no_warmup` can be used to skip the compilation stage. Thus kernels will be
+  compiled during the profiling stage. This increases tuning time. But it's
+  required for some old torch version, in which some function used in the warmup
+  kernel launch is not supported.
+- Parallel profiling of kernels: The tuning space is first divided into a number
+of tasks, which is controled by `--jobs n`. And all the tasks can be profiled in
+parallel on a number of GPUs in the system. There are two ways to specify which 
+GPU(s) we want to use for profiling. Note that these flags cannot be use together.
+By default, only one task is generated and profiled on GPU0.
+  - `--ngpus n`: GPU 0,1,.., n-1 will be used.
+  - `--gpu_ids ids`: `ids` are comma separated gpu ids and GPUs in `ids` will be used.
```

Comment:
With this PR, does it not show other GPUs as active if we only pick a subset? Previously it would use all GPUs even when a few were specified.